shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -81,12 +81,46 @@ - (void)layoutSubviews {
   [super layoutSubviews];
 }
 
+#if FLUTTER_SHELL_ENABLE_METAL
+static bool UseMetalRenderer() {
```

Comment:
If we can't make the various combinations in this method mutually exclusive, perhaps we should print some warnings for people who mess up and put conflicting ones in the project.

For example, in `--disable-metal`, check if they specified `--force-metal` or have the plist value set, and `FML_LOG(WARNING)` that we're disabling metal because this flag is set even though the other ones are.

Reply:
I was doing that but realized I was over-engineering it. This is not meant to be used by folks outside the team. In fact, this won't even be available in default builds (you still need to build locally for metal). This is just so I can test Metal in different apps without modifying the plist (test stuff like external textures, texture upload perf etc.). All command line arguments will be removed when this gets into the default builds.

Even if this were in vended builds, the only way to specify this is via command line arguments specified in Xcode schemes. That is, people can't vend binaries for others to test that depend on functionality set by command line args.

Reply:
Ohh I was mixing this up with shell arguments.

It might be nice to have a shell argument to force metal to be disabled, similar to having a shell argument to force software rendering.  

This is fine though.

Reply:
Filed https://github.com/flutter/flutter/issues/42403